File: Baekjoon/2000/2579_dp/answer.py
import sys

def solution(n, stairs):
    scores = [0, ]
    if n >= 1:
        scores.append(stairs[1])
    if n >= 2:
        scores.append(stairs[1] + stairs[2])
    if n >= 3:
        scores.append(max(stairs[1] + stairs[3], stairs[2] + stairs[3]))
    if n >= 4:
        for idx in range(4, n+1):
            scores.append(max(scores[idx-3] + stairs[idx-1] + stairs[idx], scores[idx-2] + stairs[idx]))
    return scores.pop()

if __name__ == "__main__":
    n = int(input())
    stairs = [0]
    for _ in range(1, n+1):
        stairs.append(int(input()))
    print(solution(n, stairs))

File: Programmers/lv2/42883_stack/answer.js
// 큰 수 만들기
function solution(number, k) {
  const stack = [];
  for (let num of number) {
    while (k > 0 && stack.slice(-1)[0] < num) {
      stack.pop(); //현재 number가 stack 위에있는 숫자보다 작으므로 stack에서 제거
      k--; // k는 제거해야하는 횟수; 제거 1회했으므로 1 감소
    }
    stack.push(num);
  }

  return stack.join("").substr(0, number.length - k);
}

console.log(solution("1231234", 3));
// const indexArr = Array(number.length)
//   .fill(0)
//   .map((_, i) => i);
// const cases = combination(indexArr, number.length - k);
// let max = -1;
// for (let c of cases) {
//   let now = [];
//   for (let idx of c) {
//     now.push(number[idx]);
//   }
//   max = Math.max(max, +now.sort((a, b) => a > b).join(""));
// }
// return max;
